BiometricLocker/BiometricLocker.swift

```diff
@@ -15,51 +15,80 @@ public protocol AuthenticationDelegate: class {
     func didFailAuthentication(error: LAError)
 }
 
-open class BiometricLocker {
-    public enum Key: String {
-        case applicationDidEnterBackgroundDate
+final public class BiometricLocker {
+    /// Defines the behaviour of the `lock` function.
+    ///
+    /// - now: sets the app as locked immediatelly.
+    /// - afterTimeAllowance: sets the app as locked after the `unlockedTimeAllowace` interval at checking time.
+    /// - afterTimeInterval: sets the app as locked after the given time interval at checking time.
+    public enum LockingTime {
+        case now
+        case afterTimeAllowance
+        case afterTimeInterval(TimeInterval)
+    }
+
+    /// Defines the UserDefaults key we'll use to store the date.
+    private enum Key: String {
+        case applicationDidEnterBackgroundDate = "com.bakkenbaeck.BiometricLocker.applicationDidEnterBackgroundDate"
     }
 
-    open weak var delegate: AuthenticationDelegate?
+    public weak var delegate: AuthenticationDelegate?
+
+    /**
+     Define which policy we have for the device owner to be authenticated using a biometric method (Touch ID or Face ID).
+     Defaults to `deviceOwnerAuthenticationWithBiometrics`.
 
-    open var policy: LAPolicy = .deviceOwnerAuthenticationWithBiometrics
+     See [LocalAuthentication](apple-reference-documentation://cslocalauthentication) for a more detailed explanation.
+     */
+    public var policy: LAPolicy = .deviceOwnerAuthenticationWithBiometrics
 
-    open var localizedReason: String = ""
+    /// The app-provided reason for requesting authentication, which displays in the authentication dialog presented to the user.
+    public var localizedReason: String = ""
```

Comment:
do we want to have this be passed in via the initializer instead of defaulting to an empty string, to make sure the consuming app knows they need to set this?